pkgs/development/php-packages/datadog_trace/default.nix

```
@@ -1,22 +1,56 @@
-{ buildPecl, curl, fetchFromGitHub, lib, pcre2, php }:
+{ lib
+, stdenv
+, php
+, cargo
+, rustc
+, fetchFromGitHub
+, rustPlatform
+, curl
+, pcre2
+, libiconv
+, darwin
+}@attrs:
 
-buildPecl rec {
-  pname = "ddtrace";
-  version = "0.82.0";
+php.buildPecl rec {
+  stdenv = if attrs.stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else attrs.stdenv;
+
+  pname = "datadog_trace";
+  version = "0.89.0";
 
   src = fetchFromGitHub {
     owner = "DataDog";
     repo = "dd-trace-php";
     rev = version;
-    sha256 = "sha256-QTqZRHh57mRkg0HT9qQS13emGobB0IRqM+mdImAPgtE=";
+    hash = "sha256-wTGQV80XQsBdmTQ+xaBKtFwLO3S+//9Yli9aReXDlLA=";
+    fetchSubmodules = true;
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "datadog-profiling-2.2.0" = "sha256-PWzC+E2u0hM0HhU0mgZJZvFomEJdQag/3ZK1FibSLG8=";
+    };
   };
 
-  buildInputs = [ curl pcre2 ];
+  env.NIX_CFLAGS_COMPILE = "-O2";
+
+  nativeBuildInputs = [
+    cargo
+    rustc
+    rustPlatform.bindgenHook
+    rustPlatform.cargoSetupHook
+  ];
+
+  buildInputs = [ curl pcre2 libiconv ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk_11_0.frameworks.CoreFoundation
+    darwin.apple_sdk_11_0.frameworks.Security
+  ];
```

```suggestion
  buildInputs = [ curl pcre2 ]
  ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk_11_0.frameworks.CoreFoundation
    darwin.apple_sdk_11_0.frameworks.Security
    libiconv
  ];
```
libiconv is null on linux